x/dex/keeper/core.go

```diff
@@ -283,28 +284,36 @@ func (k Keeper) WithdrawCore(goCtx context.Context, msg *types.MsgWithdrawl, tok
 }
 
 // Handles core logic for the asset 0 to asset1 direction of MsgSwap; faciliates swapping amount0 for some amount of amount1, given a specified pair (token0, token1)
-func (k Keeper) Swap0to1(goCtx context.Context, msg *types.MsgSwap, token0 string, token1 string, callerAddr sdk.AccAddress) (sdk.Int, sdk.Int, error) {
+func (k Keeper) SwapCore(goCtx context.Context,
+	msg *types.MsgSwap,
+	token0 string,
+	token1 string,
+	callerAddr sdk.AccAddress,
+	receiverAddr sdk.AccAddress,
+) (sdk.Coin, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 	pairId := CreatePairId(token0, token1)
 	feeTiers := k.GetAllFeeTier(ctx)
+	is0To1 := msg.TokenIn == token0
```

Comment:
Seems like we could introduce the idea of a "cannonical trading pair" where the two token names are sorted alphabetically (i.e. token0<>token1), and then also have an object that wraps that pair where is0To1 is stored on that object. Then this wrapped object can be passed around as a "relative pair" without breaking anything.